FifthLesson/VectorProgram/VectorProgram/ProgramRun.cs

```diff
@@ -13,87 +9,125 @@ class ProgramRun
+            Console.WriteLine(string.Empty);
+            Console.WriteLine($"{numericMultiplier} * {first} = {(numericMultiplier * first)}");
+            Console.WriteLine($"{numericMultiplier} * {second} = {(numericMultiplier * second)}");
         }
         
+        private Vector GetUserVector(string orderByVectors)
+        {
+            var userInput = RequestUserInput(DataType.Vector, $"Enter the coordinates of the {orderByVectors} three-dimensional vector separated by a space:");
+            return ParseUserInputAndCreateVector(userInput);
+        }
+
+        private double GetUserNumericMultiplier()
+        {
+            double numericInput;
+            var userInput = RequestUserInput(DataType.Multiplier, "Enter a number to produce a vector by a number:");
+
+            double.TryParse(userInput.Replace(",", "."), NumberStyles.Number, CultureInfo.CreateSpecificCulture("en-US"), out numericInput);
+            return numericInput;
+        }
+                
         private Vector ParseUserInputAndCreateVector(string userInput)
         {
-            var coords = VerificateInputData.ParseUserInput(userInput);
+            var coords = DataParser.ParseStringToArray(userInput);
             return new Vector(coords[0], coords[1], coords[2]);
         }
-
-        private string ReadInputCoordsAndVerificate()
+        
+        private string RequestUserInput(DataType dataType, string welcomMessage)
         {
-            var userInput = Console.ReadLine();
-            while(!VerificateInputData.VerifyInputDataCoordsVector(userInput))
+            var userInput = string.Empty;
+            var correctInput = false;
+
+            while (!correctInput)
             {
-                WriteErrorMessage();
-                ReadInputCoordsAndVerificate(); break;
+                Console.WriteLine(welcomMessage);
+                userInput = Console.ReadLine();
+                correctInput = ValidateUserInput(dataType, userInput);
             }
+            
             return userInput;
         }
 
-        private double ReadInputDoubleAndVerificate()
+        private bool ValidateUserInput(DataType dataType, string userInput)
         {
-            var userInput = Console.ReadLine();
-            double numericInput;
-            while (!VerificateInputData.VerifyInputDataCoordsVector(userInput))
-            {
-                WriteErrorMessage();
-                ReadInputDoubleAndVerificate();
-            }
-            double.TryParse(userInput.Replace(",", "."), NumberStyles.Number, CultureInfo.CreateSpecificCulture("en-US"), out numericInput);
-            return numericInput;
+            var correctInput = false;
+            correctInput = Validator.ValidateInput(dataType, userInput);
+            if (!correctInput) WriteErrorMessage();
+            return correctInput;
         }
+        
+        private void WriteErrorMessage() => Console.WriteLine("You entered non-correct numbers.");
```

Comment:
Maybe it would be better to say that user entered **incorrect value**?

Reply:
Maybe it would be better to say that user entered incorrect value?